integration-cli/docker_api_update_unix_test.go

```diff
@@ -34,3 +34,40 @@ func (s *DockerSuite) TestAPIUpdateContainer(c *check.C) {
 	out, _ = dockerCmd(c, "exec", name, "cat", file)
 	c.Assert(strings.TrimSpace(out), checker.Equals, "524288000")
 }
+
+func (s *DockerSuite) TestAPIUpdateContainerPIDsLimit(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+	testRequires(c, pidsLimit)
+
+	name := "apiUpdateContainerPIDsLimit"
+	hostConfig := map[string]interface{}{
+		"PidsLimit": 15,
+	}
+	dockerCmd(c, "run", "-d", "--name", name, "--pids-limit", "20", "busybox", "top")
+	c.Assert(inspectField(c, name, "HostConfig.PidsLimit"), checker.Equals, "20")
+	_, _, err := request.SockRequest("POST", "/containers/"+name+"/update", hostConfig, daemonHost())
+	c.Assert(err, check.IsNil)
+
+	c.Assert(inspectField(c, name, "HostConfig.PidsLimit"), checker.Equals, "15")
+
+	file := "/sys/fs/cgroup/pids/pids.max"
+	out, _ := dockerCmd(c, "exec", name, "cat", file)
+	c.Assert(strings.TrimSpace(out), checker.Equals, "15")
+}
+
+func (s *DockerSuite) TestAPIContainerPidsLimit(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+	testRequires(c, pidsLimit)
+
+	name := "apiContainerPidsLimit"
+	dockerCmd(c, "run", "-d", "--name", name, "--pids-limit", "5", "busybox", "top")
+	c.Assert(inspectField(c, name, "HostConfig.PidsLimit"), checker.Equals, "5")
+
+	dockerCmd(c, "exec", name, "top")
+	dockerCmd(c, "exec", name, "top")
+	dockerCmd(c, "exec", name, "top")
+	dockerCmd(c, "exec", name, "top")
+
+	_, _, err := dockerCmdWithError("exec", name, "top")
+	c.Assert(err, checker.NotNil)
```

Comment:
I'm not sure what's wrong here. Whenever I use `dockerCmdWithError`, the test gets stuck and I have to stop it or it times out after a long time. idk why it's not working here, when other tests are using this in the same way :(